packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -456,6 +464,11 @@ export class PipelineEditor extends React.Component<
     }
     const app_data = node.app_data;
 
+    if (app_data.filename !== propertySet.filename) {
+      app_data.filename = propertySet.filename;
```

Comment:
Note this would specifically need to be updated to the following once #861 is merged or in #861 once this is merged.
```suggestion
    const propertyFilename = PipelineService.getPipelineRelativeNodePath(
      this.widgetContext.path,
      propertySet.filename
    );
    if (app_data.filename !== propertyFilename) {
      app_data.filename = propertyFilename;
```